### PR TITLE
test(amplify-console-integration): consolidate Console related tests

### PR DESCRIPTION
**Pull request by UnleashedMind**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-console-integration/projects-templates/create-react-app-auth-amplify/CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 ## Code of Conduct
+
```
Comment (yuth) [inline packages/amplify-console-integration/projects-templates/create-react-app-auth-amplify/CODE_OF_CONDUCT.md]: why?

File: packages/amplify-console-integration/package.json [hunk 2]
```diff
@@ -59,7 +59,15 @@
     ],
     "reporters": [
       "default",
-      "jest-junit"
+      "jest-junit",
```
Comment (yuth) [inline packages/amplify-console-integration/package.json]: can we name this package as `amplify-console-integration-tests` so its clear this is a test package

File: packages/amplify-console-integration/__tests__/consoleHosting.test.ts [hunk 3]
```diff
@@ -47,7 +57,7 @@ describe('amplify console add hosting', () => {
 
   it('when hosting is enabled, add new env should be able to deploy frontend successfully', async () => {
     await addManualHosting(projRoot);
-    await addEnvironment(projRoot, { envName: NWE_ENV });
+    await addEnvironment(projRoot, { providersParam, envName: NWE_ENV });
```
Comment (yuth) [inline packages/amplify-console-integration/__tests__/consoleHosting.test.ts]: rename `NWE_ENV` -> `NEW_ENV`

File: packages/amplify-console-integration/projects-templates/create-react-app-auth-amplify/CONTRIBUTING.md [hunk 4]
```diff
@@ -6,24 +6,23 @@ documentation, we greatly value feedback and contributions from our community.
...
 
 When filing an issue, please check [existing open](https://github.com/aws-samples/create-react-app-auth-amplify/issues), or [recently closed](https://github.com/aws-samples/create-react-app-auth-amplify/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), issues to make sure somebody else hasn't already
 reported the issue. Please try to include as much information as you can. Details like these are incredibly useful:
 
-* A reproducible test case or series of steps
-* The version of our code being used
-* Any modifications you've made relevant to the bug
-* Anything unusual about your environment or deployment
-
+- A reproducible test case or series of steps
```
Comment (yuth) [inline packages/amplify-console-integration/projects-templates/create-react-app-auth-amplify/CONTRIBUTING.md]: Why is this file included? I dont see any changes here

File: packages/amplify-console-integration/projects-templates/create-react-app-auth-amplify/README.md [hunk 5]
```diff
@@ -1,4 +1,4 @@
-# Create-react-app with AWS Amplify Auth 
+# Create-react-app with AWS Amplify Auth
```
Comment (yuth) [inline packages/amplify-console-integration/projects-templates/create-react-app-auth-amplify/README.md]: Same with this file

File: packages/amplify-console-integration/projects-templates/create-react-app-auth-amplify/public/index.html [hunk 6]
```diff
@@ -1,15 +1,15 @@
 <!DOCTYPE html>
 <html lang="en">
   <head>
-    <meta charset="utf-8">
-    <link rel="shortcut icon" href="%PUBLIC_URL%/favicon.ico">
-    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
-    <meta name="theme-color" content="#000000">
+    <meta charset="utf-8" />
```
Comment (yuth) [inline packages/amplify-console-integration/projects-templates/create-react-app-auth-amplify/public/index.html]: Didnt we decide not to ship the project template as part of our repo?

File: packages/amplify-console-integration/src/pullAndInit/pullProject.ts [hunk 7]
```diff
@@ -1,20 +1,19 @@
-import * as nexpect from 'nexpect';
+// import * as nexpect from 'nexpect';
```
Comment (yuth) [inline packages/amplify-console-integration/src/pullAndInit/pullProject.ts]: remove

File: packages/amplify-console-integration/__tests__/consoleHosting.test.ts (hunk 3 above)
Comment (UnleashedMind) [reply packages/amplify-console-integration/__tests__/consoleHosting.test.ts]: check

File: packages/amplify-console-integration/projects-templates/create-react-app-auth-amplify/CODE_OF_CONDUCT.md (hunk 1 above)
Comment (UnleashedMind) [reply packages/amplify-console-integration/projects-templates/create-react-app-auth-amplify/CODE_OF_CONDUCT.md]: why the added empty line? 
it was not manually added, but introduced by `npm run commit`

File: packages/amplify-console-integration/src/pullAndInit/pullProject.ts (hunk 7 above)
Comment (UnleashedMind) [reply packages/amplify-console-integration/src/pullAndInit/pullProject.ts]: check

File: packages/amplify-console-integration/projects-templates/create-react-app-auth-amplify/public/index.html (hunk 6 above)
Comment (UnleashedMind) [reply packages/amplify-console-integration/projects-templates/create-react-app-auth-amplify/public/index.html]: I don't recall. 
I guess Gary included it to avoid having to include `create-react-app` as a dependency and the lengthy testing project creation time .

File: packages/amplify-console-integration/package.json (hunk 2 above)
Comment (UnleashedMind) [reply packages/amplify-console-integration/package.json]: check

File: packages/amplify-console-integration/package.json [hunk 8]
```diff
@@ -20,27 +20,27 @@
...
+    "amplify-e2e-core": "1.0.1",
     "@types/ini": "^1.3.30",
     "aws-sdk": "^2.608.0",
     "dotenv": "^8.2.0",
     "esm": "^3.2.25",
     "fs-extra": "^8.1.0",
+    "chalk": "^3.0.0",
     "ini": "^1.3.5",
     "moment": "^2.24.0",
     "nexpect": "^0.6.0",
```
Comment (yuth) [inline packages/amplify-console-integration/package.json]: we dont need it anymore

File: packages/amplify-console-integration/projects-templates/create-react-app-auth-amplify/CODE_OF_CONDUCT.md (hunk 1 above)
Comment (UnleashedMind) [reply packages/amplify-console-integration/projects-templates/create-react-app-auth-amplify/CODE_OF_CONDUCT.md]: removed the file

File: packages/amplify-console-integration/projects-templates/create-react-app-auth-amplify/CONTRIBUTING.md (hunk 4 above)
Comment (UnleashedMind) [reply packages/amplify-console-integration/projects-templates/create-react-app-auth-amplify/CONTRIBUTING.md]: removed the file

File: packages/amplify-console-integration/projects-templates/create-react-app-auth-amplify/public/index.html (hunk 6 above)
Comment (yuth) [reply packages/amplify-console-integration/projects-templates/create-react-app-auth-amplify/public/index.html]: We should not be including anything we dont maintain as part of our tests. The code can go out of sync and our tests would not be testing the things customer is doing. So lets not ship this part of our repo.

File: packages/amplify-console-integration/package.json (hunk 8 above)
Comment (UnleashedMind) [reply packages/amplify-console-integration/package.json]: check

Comment (UnleashedMind) [reply packages/amplify-console-integration/package.json]: removed

File: packages/amplify-console-integration/projects-templates/create-react-app-auth-amplify/CONTRIBUTING.md (hunk 4 above)
Comment (UnleashedMind) [reply packages/amplify-console-integration/projects-templates/create-react-app-auth-amplify/CONTRIBUTING.md]: removed

File: packages/amplify-console-integration/projects-templates/create-react-app-auth-amplify/README.md (hunk 5 above)
Comment (UnleashedMind) [reply packages/amplify-console-integration/projects-templates/create-react-app-auth-amplify/README.md]: I think it's good to keep this read me file, so readers know that the auth starter sample project is used in the console hosting e2e test.

**Comment by UnleashedMind** [comment]
We can not just run the Console integration tests periodically, because PR changes can break them, and result broken Console experience.

File: jest.config.js [hunk 9]
```diff
@@ -10,7 +10,7 @@ module.exports = {
     '**/__integration__/',
     'packages/amplify-e2e-core/',
     'packages/amplify-e2e-tests/',
-    'packages/amplify-console-integration/',
+    'packages/amplify-console-integration-tests/',
```
Comment (yuth) [inline jest.config.js]: We dont need this here. Otherwise these tests will be run by jest-vscode plugin and would end up clogging your account

File: packages/amplify-console-integration-tests/__tests__/consoleHosting.test.ts [hunk 10]
```diff
@@ -36,7 +46,7 @@ describe('amplify console add hosting', () => {
       expect(fs.existsSync(path.join(projRoot, 'amplify', 'backend', 'hosting', 'amplifyhosting'))).toBe(true);
       const type = loadTypeFromTeamProviderInfo(projRoot, ORIGINAL_ENV);
       expect(type).toBe(TYPE_MANUAL);
-      await npmInstall(projRoot);
+      npmInstall(projRoot);
```
Comment (yuth) [inline packages/amplify-console-integration-tests/__tests__/consoleHosting.test.ts]: is this not an async method?

File: packages/amplify-console-integration-tests/src/consoleHosting/consoleHosting.ts [hunk 11]
```diff
@@ -1,6 +1,89 @@
 import { nspawn as spawn } from 'amplify-e2e-core';
-import * as nexpect from 'nexpect';
-import { getCLIPath } from '../../utils';
+import { getCLIPath } from '../util';
```
Comment (yuth) [inline packages/amplify-console-integration-tests/src/consoleHosting/consoleHosting.ts]: This method should be moved to amplify-e2e-core as well

File: packages/amplify-console-integration-tests/src/consoleHosting/consoleHosting.ts [hunk 12]
```diff
@@ -118,15 +201,7 @@ export function amplifyPush(cwd: string) {
...
-  return new Promise((resolve, reject) => {
-    nexpect.spawn('npm install', { cwd, stripColors: true }).run((err: Error) => {
-      if (!err) {
-        resolve();
-      } else {
-        reject(err);
-      }
-    });
-  });
+  spawnSync('npm', ['install'], { cwd });
```
Comment (yuth) [inline packages/amplify-console-integration-tests/src/consoleHosting/consoleHosting.ts]: this should be moved to e2e-core as we have the same copy in e2e-tests

File: packages/amplify-console-integration-tests/src/consoleHosting/utils.ts [hunk 13]
```diff
@@ -34,3 +36,12 @@ export function loadAppIdFromTeamProviderInfo(cwd: string, currEnv: string) {
 function readJsonFile(jsonFilePath: string, encoding = 'utf8') {
   return JSON.parse(fs.readFileSync(jsonFilePath, encoding));
 }
+
+export function createTestProject(): string {
+  const projectName = `amplify-integ-${Math.round(Math.random() * 100)}-test-${Math.round(Math.random() * 1000)}`;
```
Comment (yuth) [inline packages/amplify-console-integration-tests/src/consoleHosting/utils.ts]: we should move the similar function we have in amplify-e2e-tests to amplify-e2e-core and reuse it instead of creating a new one here

File: packages/amplify-console-integration-tests/src/profile-helper.ts [hunk 14]
```diff
@@ -0,0 +1,101 @@
+import { getProfileName } from './util';
```
Comment (yuth) [inline packages/amplify-console-integration-tests/src/profile-helper.ts]: Dont we have a similar method in amplify-e2e-tests. Lets have only one implementation which is a superset of features and move it to amplify-e2e-core

**Comment by UnleashedMind** [comment]
There's on issue with LGTM comments, the PR's last commit fixed an alert because it deleted a file that had unused import.

File: jest.config.js (hunk 9 above)
Comment (UnleashedMind) [reply jest.config.js]: is that exactly why we need to put it into the `testPathIgnorePatterns`?

File: packages/amplify-console-integration-tests/__tests__/consoleHosting.test.ts (hunk 10 above)
Comment (UnleashedMind) [reply packages/amplify-console-integration-tests/__tests__/consoleHosting.test.ts]: no, it's run by spawnSync

File: packages/amplify-console-integration-tests/src/consoleHosting/consoleHosting.ts (hunk 11 above)
Comment (UnleashedMind) [reply packages/amplify-console-integration-tests/src/consoleHosting/consoleHosting.ts]: moved

File: packages/amplify-console-integration-tests/src/consoleHosting/consoleHosting.ts (hunk 12 above)
Comment (UnleashedMind) [reply packages/amplify-console-integration-tests/src/consoleHosting/consoleHosting.ts]: moved

File: packages/amplify-console-integration-tests/src/consoleHosting/utils.ts (hunk 13 above)
Comment (UnleashedMind) [reply packages/amplify-console-integration-tests/src/consoleHosting/utils.ts]: the name should be changed to indicate it is a console test.

File: packages/amplify-console-integration-tests/src/profile-helper.ts (hunk 14 above)
Comment (UnleashedMind) [reply packages/amplify-console-integration-tests/src/profile-helper.ts]: The profile Name is different in the console-integration-tests, this profile will use the aws account specially set up for the Amplify Console related tests.

File: jest.config.js (hunk 9 above)
Comment (yuth) [reply jest.config.js]: last I checked testIgnorePattern did not seemed to work for monrepo. Can you confirm ? If it doesnt please remove this from here.

File: packages/amplify-console-integration-tests/src/consoleHosting/consoleHosting.ts (hunk 11 above)
Comment (yuth) [reply packages/amplify-console-integration-tests/src/consoleHosting/consoleHosting.ts]: I dont see the moved code. It still is referencing `../util`

File: packages/amplify-console-integration-tests/src/profile-helper.ts (hunk 14 above)
Comment (yuth) [reply packages/amplify-console-integration-tests/src/profile-helper.ts]: we could make the helper function accept the profile name and regions then, instead of having multiple copies and making it harder to maintain

File: .eslintrc.js [hunk 15]
```diff
@@ -203,7 +203,7 @@ module.exports = {
 
     '/packages/amplify-graphql-types-generator/test', // Not linting tests yet
     'amplify-e2e-tests', // Not linting test projects yet
-    'amplify-console-integration',
+    'amplify-console-integration-tests',
```
Comment (yuth) [inline .eslintrc.js]: Since this is new package we should not add it here in lint ignore. We have e2e tests here because lint was enabled later. Our goal is to make sure everything passes the linter

File: packages/amplify-console-integration-tests/src/util.ts [hunk 16]
```diff
@@ -2,6 +2,8 @@ import * as path from 'path';
 import * as fs from 'fs-extra';
 import * as moment from 'moment';
 
+export { getCLIPath, isCI, npmInstall } from 'amplify-e2e-core';
+
 export function createNewProjectDir(mark?: string): string {
```
Comment (yuth) [inline packages/amplify-console-integration-tests/src/util.ts]: We could reuse this by moving it to e2e core and have parity as how we name our tests. Just naming them `console-inge` would make it harder to track which test is is not cleaning up the resources, when the resources remain in the account.  We also have had issues with test name collisions as the granularity of the  test name is seconds and not milli or nano seconds.
https://github.com/aws-amplify/amplify-cli/blob/c6ea964712d1fc127466822638a076c7492260ab/packages/amplify-e2e-tests/src/utils/index.ts#L30-L42

**Comment by UnleashedMind** [comment]
LGTM is all green with a unused variable fixed alert
https://github.com/aws-amplify/amplify-cli/pull/3749/checks?check_run_id=541169384

1 fixed alert
This pull request fixes 1 alert when merging 9ccc450 into af90f56 - view on LGTM.com

fixed alerts:

1 for Unused variable, import, function or class

File: packages/amplify-console-integration-tests/src/consoleHosting/consoleHosting.ts (hunk 11 above)
Comment (UnleashedMind) [reply packages/amplify-console-integration-tests/src/consoleHosting/consoleHosting.ts:2]: https://github.com/aws-amplify/amplify-cli/pull/3749/files#diff-36366797da092f7d5734749205990a8eR5

File: jest.config.js (hunk 9 above)
Comment (yuth) [reply jest.config.js:13]: My Bad, I did not realize this is in test-ignore pattern
